config/core/configmaps/deployment.yaml

```diff
@@ -91,3 +91,20 @@ data:
     # Sets rootCA for the queue proxy - used by QPOptions
     # If omitted, or empty, no rootCA is added to the golang rootCAs
     queue-sidecar-rootca: ""
+
+    # If set to "true", it automatically configures pod anti-affinity requirements for all Knative services.
+    # It employs the `preferredDuringSchedulingIgnoredDuringExecution` weighted pod affinity term,
+    # aligning with the Knative revision label. It yields the configuration below in all workloads' deployments:
+    # `
+    #   affinity:
+    #     podAntiAffinity:
+    #       preferredDuringSchedulingIgnoredDuringExecution:
+    #       - podAffinityTerm:
+    #           topologyKey: kubernetes.io/hostname
+    #           labelSelector:
+    #             matchLabels:
+    #               serving.knative.dev/revision: {{revision-name}}
+    #         weight: 100
+    # `
+    #
+    # enable-pod-anti-affinity-rule: "true"
```

Comment:
Instead of a `bool` we should take a literal string to represent the affinity setting we want.

This would make the option extensible to future new options.

eg. disable this option

```
affinity: none
```

eg. revision instances are split across nodes
```
affinity: prefer-spread-revision-over-nodes
```

Reply:
@dprotaso I've addressed the feedback